src/esm_environment/esm_environment.py

```diff
@@ -525,11 +541,66 @@ def get_shell_commands(self):
         environment.append("")
         # Write the unset commands
         if self.config.get("unset_vars") is not None:
+            self.sort_env_vars("unset_vars", esm_parser.CATEGORY_HIERARCHY)
             for var in self.config["unset_vars"]:
                 environment.append(f"unset {var}")
 
         return environment
 
+    def select_env_vars_based_on_provenance(self, env_var_key):
+        pass
+
+    def select_env_vars_based_on_var_attributes(self, env_var_key):
+        pass
+
+    def sort_env_vars(self, env_var_key, category_order):
```

Comment:
I'll try to write docs based on my understanding, please improve or amend as necessary:

```suggestion
    def sort_env_vars(self, env_var_key, category_order):
        """"
        Sorts environment variables based upon their order in the original configuration files, respecting their provenance.
        
        Parameters
        ----------
        env_var_key : str
            The environment variable to be sorted (e.g. ``LD_LIBRARY_PATH``)
        category_order : list
            The category priorities to respect, i.e. ``["computer", "component", "setup"].
            
        Mutates
        -------
        self.config : dict
           The EnvironmentInfo ``config`` dictionary is modified specifically for
           ``env_var_key``, with the resorted, provenance-aware order.
        """
```